crates/re_viewer/src/app_state.rs

```diff
@@ -17,16 +18,16 @@ const WATERMARK: bool = false; // Nice for recording media material
 #[serde(default)]
 pub struct AppState {
     /// Global options for the whole viewer.
-    app_options: AppOptions,
+    app_options: RwLock<AppOptions>,
 
     /// Things that need caching.
     #[serde(skip)]
     pub(crate) cache: Caches,
 
     #[serde(skip)]
-    selected_rec_id: Option<StoreId>,
+    selected_rec_id: RwLock<Option<StoreId>>,
     #[serde(skip)]
-    pub(crate) selected_blueprint_by_app: HashMap<ApplicationId, StoreId>,
+    selected_blueprint_by_app: RwLock<HashMap<ApplicationId, StoreId>>,
```

Comment:
I'm not sure this is a good solution. It's nice that `&AppState` means read-only, and adding interior mutability to everything is quite unidiomatic, and an ugly way to circumvent the borrowchecker when we should not need to circumvent it at all